crates/bevy_render/Cargo.toml

```diff
@@ -57,7 +57,8 @@ bevy_tasks = { path = "../bevy_tasks", version = "0.12.0-dev" }
 image = { version = "0.24", default-features = false }
 
 # misc
-wgpu = { version = "0.16.0", features=["naga"] }
+wgpu = { version = "0.16.0", features = ["naga"] }
+wgpu-types = { version = "0.16.0", features = ["trace", "replay"] }
```

Comment:
Why is `wgpu-types` needed?

Reply:
I added it to  enable the `trace` and `replay` features unconditionally only for the types re-exported in `wgpu` from there (all types used in the fields of `wgpu::SamplerDescriptor`), without enabling the `trace`/`replay` related stuff in `wgpu` itself. But I can alos move the enabled features to `wgpu`.

Reply:
Do we need to enable these if we're using a proxy type? `wgpu` is already pretty slow to compile as is, so I'd prefer to avoid making it take even longer by enabling serde.

Reply:
It is still needed since the proxy type is used for the `Serialize`/`Deserialize` impl of `wgpu::SamplerDescriptor`, but the `Serialize`/`Deserialize` impl of its fields is pulled from the serde derives in `wgpu_types`. This should only impact the compile time of `wgpu_types` which is just a collection of types re-exported in `wgpu`, from some quick testing the compile times of `wgpu` itself aren't affected, and `wgpu_types` goes from 1.2s to 1.7s to build when those features are enabled.

Reply:
Fwiw if we're going with this solution I'd rather we go with declaring a dependency on `wgpu-types` and enable the serde features on it (trace+replay), vs enabling them for the whole wgpu crate. (wgpu always includes wgpu-types, so this isn't adding another dependency, it's just not slowing down our compiles as much as enabling trace+replay for `wgpu` would).